src/System.Private.CoreLib/shared/System/Progress.cs

```diff
@@ -34,7 +34,7 @@ public Progress()
             // If there is no current context, we use a default instance targeting the ThreadPool.
             _synchronizationContext = SynchronizationContext.Current ?? ProgressStatics.DefaultContext;
             Debug.Assert(_synchronizationContext != null);
-            _invokeHandlers = new SendOrPostCallback(InvokeHandlers);
+            _invokeHandlers = new SendOrPostCallback(InvokeHandlers!); // TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/35896
```

Comment:
We should remove the TODO, remove the !, and change `InvokeHandlers` to take an `object?`, then change `(T)state` to `(T)state!`.  null can actually be passed in, but only if T is nullable, so it makes sense to suppress the warning at the place where it's actually doing the conversion that's known to be safe.

Reply:
Done